tests/functional/test_config_groups.py

```diff
@@ -0,0 +1,152 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=redefined-outer-name, unused-argument, duplicate-code
+
+from typing import List
+
+import allure
+import pytest
+from adcm_client.objects import (
+    ADCMClient,
+    Cluster,
+    Provider,
+    Host,
+)
+from adcm_pytest_plugin import utils
+from adcm_pytest_plugin.utils import get_data_dir
+from coreapi.exceptions import ErrorMessage
+
+from tests.library.errorcodes import GROUP_CONFIG_HOST_ERROR
+
+CLUSTER_BUNDLE_PATH = get_data_dir(__file__, "cluster_simple")
+PROVIDER_BUNDLE_PATH = get_data_dir(__file__, "hostprovider_bundle")
+
+
+@pytest.fixture()
+def cluster(request, sdk_client_fs: ADCMClient) -> Cluster:
+    bundle_path = request.param if hasattr(request, "param") else CLUSTER_BUNDLE_PATH
+    cluster_bundle = sdk_client_fs.upload_from_fs(bundle_path)
+    return cluster_bundle.cluster_create(name=utils.random_string())
+
+
+@pytest.fixture()
+def provider(request, sdk_client_fs: ADCMClient) -> Provider:
+    bundle_path = request.param if hasattr(request, "param") else PROVIDER_BUNDLE_PATH
+    provider_bundle = sdk_client_fs.upload_from_fs(bundle_path)
+    return provider_bundle.provider_create(name=utils.random_string())
+
+
+class TestGroupsIntersection:
+
+    HOST_ERROR_MESSAGE = (
+        "host is not available for this object, or host already is a member of another group of this object"
+    )
+    FIRST_COMPONENT_NAME = "first"
+    SECOND_COMPONENT_NAME = "second"
+    FIRST_GROUP = "test_group"
+    SECOND_GROUP = "test_group_2"
+    FIRST_HOST = "test_host_1"
+    SECOND_HOST = "test_host_2"
+
+    def create_two_hosts(self, provider: Provider):
+        with allure.step("Create host for config groups"):
+            test_host_1 = provider.host_create(fqdn=self.FIRST_HOST)
+        with allure.step("Create host for host candidate check"):
+            test_host_2 = provider.host_create(fqdn=self.SECOND_HOST)
+        return test_host_1, test_host_2
+
+    @allure.step('Check error')
+    def assert_that_host_add_is_unavaliable(self, error):
+        GROUP_CONFIG_HOST_ERROR.equal(error)
```

Comment:
Can you make it more verbose what error is expected? For example, add it to the step message and add typing for arguments

Reply:
The message will look like:
Expected error code is "GROUP_CONFIG_HOST_ERRORs", actual is "GROUP_CONFIG_HOST_ERROR"
isn't it enought?
error has type _pytest._code.ExceptionInfo[_E]

Reply:
It will be an error message, but the step message tells us what we expect even when everything is alright.
Maybe it's not the use case for error-checking, but otherwise, how do we tell what are we checking if the test is green and someone asks us about it?

Reply:
added steps